src/terrain/draw_terrain_raster.js

```diff
@@ -174,6 +174,8 @@ function drawTerrainRaster(painter: Painter, terrain: Terrain, sourceCache: Sour
 }
 
 function drawTerrainDepth(painter: Painter, terrain: Terrain, sourceCache: SourceCache, tileIDs: Array<OverscaledTileID>) {
+    assert(painter.renderPass === 'offscreen');
```

Comment:
Moved depth drawing in offscreen pass to reduce switching framebuffer from main as suggested by @astojilj .